elytron/src/main/java/org/wildfly/extension/elytron/KeyStoreService.java

```diff
@@ -108,6 +109,27 @@ static KeyStoreService createFileBasedKeyStoreService(String provider, String ty
     @Override
     public void start(StartContext startContext) throws StartException {
         try {
+
+            if (ElytronDescriptionConstants.AUTOMATIC.equalsIgnoreCase(type)) {
+                pathResolver = pathResolver();
```

Comment:
Lot of lines from added code is duplication to main code flow. Have to be?

Reply:
cleaned up